### PR TITLE
Update to use proper WASM identifiers

### PR DESCRIPTION
**Pull request by ehildenb**
This isn't a perfect match of allowable identifiers, but I was having trouble getting the Flex lexer to accept the remaining characters. The unsupported characters are documented, so that a later PR can come through and fix them.

@mrchico I know you were looking at this, can you take a peek and see what you think?

### CODE REVIEW

**Comment by MrChico** [comment]
I don't think wasm supports straight up Strings as function identifiers, but I guess you already know that and are allowing it intentionally for now?

**Comment by ehildenb** [comment]
Hmmm, these commits are from way long ago, so maybe my thinking at the time was that I didn't want to change over all the test-sets. I'll double-check the spec and remove the strings as an option if it's the case.

**Comment by ehildenb** [comment]
@MrChico I've updaded to remove support for `String` as a function name, if you'd like to double-check.

**Comment by MrChico** [comment]
LGTM!

File: data.md [hunk 1]
```diff
@@ -187,3 +201,8 @@ Operator `_++_` implements an append operator for sort `Stack`.
     rule #drop(TYPE VTYPES, < TYPE > VAL:Number : STACK) => #drop(VTYPES, STACK)
 endmodule
 ```
+
+Resources
+=========
+
+[WASM Spec]: <https://github.com/WebAssembly/design>
```
Comment (hjorthjort) [inline data.md]: The design repo is a bit scattered. Maybe use the spec repo instead, or link directly to the spec, fingers crossed they don't change the URL:
https://webassembly.github.io/spec/core/text/values.html#text-id

File: data.md [hunk 2]
```diff
@@ -8,17 +8,31 @@ module WASM-DATA
...
+
+**TODO**: Unsupported characters: `.:^@`
 
 ```k
-    syntax #Layout ::= r"(\\(;([^;]|(;+([^;\\)])))*;\\))"
-                     | r"(;;[^\\n\\r]*)"
-                     | r"([\\ \\n\\r\\t])"
- // --------------------------------------
+    syntax Identifier ::= r"\\$[0-9a-zA-Z!$%&'*+/<>?_`|~=-]*" [avoid, token]
+ // ------------------------------------------------------------------------
```
Comment (hjorthjort) [inline data.md]: One `-` too many.

File: wasm.md [hunk 3]
```diff
@@ -428,8 +428,8 @@ Function declarations can look quite different depending on which fields are omm
 Here, we allow for an "abstract" function declaration using syntax `func_::___`, and a more concrete one which allows arbitrary order of declaration of parameters, locals, and results.
 
 ```k
-    syntax FunctionName ::= Int | String
- // ------------------------------------
+    syntax FunctionName ::= Int | Identifier
+ // ----------------------------------------
```
Comment (hjorthjort) [inline wasm.md]: Again, one `-` too many.

File: data.md (hunk 2 above)
Comment (ehildenb) [reply data.md]: Hmmm, these show up as correct on mine. Can you double-check locally in your editor?

File: data.md (hunk 1 above)
Comment (ehildenb) [reply data.md]: Updated to use https://webassembly.github.io/spec

(so users can choose which version to view)

File: data.md (hunk 2 above)
Comment (hjorthjort) [reply data.md]: Yep, they are correct.
Looks like GitHub contracts the `fi` in `Identifier` when displaying.

**Pull request merged**